### ISSUE TITLE
Build Errors with Visual Studio 2022

### ISSUE THREAD

**Issue by vashero** [SaschaWillems/VulkanInfiniteProceduralTerrain#4]
I get the following errors and warnings:
PS C:\Users\<user>\desktop\VulkanInfiniteProceduralTerrain> cmake --build build/windows --config Release --target VulkanInfiniteProceduralTerrain                                                                                                MSBuild version 17.10.4+10fbfbf2e for .NET Framework

  1>Checking Build System
  Generating debug_csm.frag.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/debug_csm.frag
  Generating debug_csm.vert.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/debug_csm.vert
  Generating depthpass.vert.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/depthpass.vert
  Generating grass.frag.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/grass.frag
  Generating grass.vert.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/grass.vert
  Generating quad.frag.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/quad.frag
  Generating quad.vert.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/quad.vert
  Generating skysphere.frag.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/skysphere.frag
  Generating skysphere.vert.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/skysphere.vert
  Generating terrain.frag.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/terrain.frag
  Generating terrain.vert.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/terrain.vert
  Generating terrain_depthpass.frag.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/terrain_depthpass.frag
  Generating tree.frag.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/tree.frag
  Generating tree.vert.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/tree.vert
  Generating tree_depthpass.frag.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/tree_depthpass.frag
  Generating tree_depthpass.vert.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/tree_depthpass.vert
  Generating tree_single.vert.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/tree_single.vert
  Generating uioverlay.frag.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/base/uioverlay.frag
  Generating uioverlay.vert.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/base/uioverlay.vert
  Generating water.frag.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/water.frag
  Generating water.vert.spv
  C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/water.vert
  Building Custom Rule C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/data/shaders/CMakeLists.txt
  Building Custom Rule C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/external/CMakeLists.txt
  texture.c
  hashlist.c
  checkheader.c
  swap.c
  memstream.c
  filestream.c
  Generating Code...
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\external\ktx\lib\texture.c(31,9): warning C4005: '_CRT_SECURE_NO
_WARNINGS': macro redefinition [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\external\ktx.vcxpr
oj]
  ktx.vcxproj -> C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\external\Release\ktx.lib
  Building Custom Rule C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/base/CMakeLists.txt
  imgui.cpp
  imgui_demo.cpp
  imgui_draw.cpp
  imgui_widgets.cpp
  Noise.cpp
  VulkanAndroid.cpp
  VulkanDebug.cpp
  VulkanExampleBase.cpp
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\base\VulkanExampleBase.cpp(2225,13): warning C4244: 'initializin
g': conversion from 'T' to 'int32_t', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\bui
ld\windows\base\base.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\base\VulkanExampleBase.cpp(2225,13): warning C4244:         with
 [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\base\base.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\base\VulkanExampleBase.cpp(2225,13): warning C4244:         [ [C
:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\base\base.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\base\VulkanExampleBase.cpp(2225,13): warning C4244:
T=float [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\base\base.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\base\VulkanExampleBase.cpp(2225,13): warning C4244:         ] [C
:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\base\base.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\base\VulkanExampleBase.cpp(2226,13): warning C4244: 'initializin
g': conversion from 'T' to 'int32_t', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\bui
ld\windows\base\base.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\base\VulkanExampleBase.cpp(2226,13): warning C4244:         with
 [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\base\base.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\base\VulkanExampleBase.cpp(2226,13): warning C4244:         [ [C
:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\base\base.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\base\VulkanExampleBase.cpp(2226,13): warning C4244:
T=float [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\base\base.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\base\VulkanExampleBase.cpp(2226,13): warning C4244:         ] [C
:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\base\base.vcxproj]
  VulkanTools.cpp
  VulkanUIOverlay.cpp
  VulkanglTFModel.cpp
  Generating Code...
  base.vcxproj -> C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\base\Release\base.lib
  Building Custom Rule C:/Users/<user>/Desktop/VulkanInfiniteProceduralTerrain/src/CMakeLists.txt
  AssetManager.cpp
  HeightMapSettings.cpp
  InfiniteTerrain.cpp
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\base\VulkanTexture.hpp(830,57): warning C4267: '=': conversion f
rom 'size_t' to 'uint32_t', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows
\src\VulkanInfiniteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\base\VulkanHeightmap.hpp(129,22): warning C4305: 'initializing':
 truncation from 'double' to 'float' [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\src\VulkanIn
finiteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\InfiniteTerrain.cpp(13,37): warning C4244: '=': conversion f
rom 'float' to 'int', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\src\V
ulkanInfiniteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\InfiniteTerrain.cpp(17,37): warning C4244: '=': conversion f
rom 'float' to 'int', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\src\V
ulkanInfiniteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\InfiniteTerrain.cpp(52,18): warning C4244: 'initializing': c
onversion from 'float' to 'int', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\wi
ndows\src\VulkanInfiniteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\InfiniteTerrain.cpp(53,18): warning C4244: 'initializing': c
onversion from 'float' to 'int', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\wi
ndows\src\VulkanInfiniteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\InfiniteTerrain.cpp(65,31): warning C4244: 'initializing': c
onversion from 'float' to 'int', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\wi
ndows\src\VulkanInfiniteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\InfiniteTerrain.cpp(65,24): warning C4244: 'initializing': c
onversion from 'float' to 'const int', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\bu
ild\windows\src\VulkanInfiniteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\InfiniteTerrain.cpp(66,31): warning C4244: 'initializing': c
onversion from 'float' to 'int', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\wi
ndows\src\VulkanInfiniteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\InfiniteTerrain.cpp(66,24): warning C4244: 'initializing': c
onversion from 'float' to 'const int', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\bu
ild\windows\src\VulkanInfiniteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\InfiniteTerrain.cpp(69,123): warning C4244: 'argument': conv
ersion from 'float' to 'int', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windo
ws\src\VulkanInfiniteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\InfiniteTerrain.cpp(69,74): warning C4244: 'argument': conve
rsion from 'float' to 'int', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\window
s\src\VulkanInfiniteProceduralTerrain.vcxproj]
  TerrainChunk.cpp
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\base\VulkanTexture.hpp(830,57): warning C4267: '=': conversion f
rom 'size_t' to 'uint32_t', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows
\src\VulkanInfiniteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\base\VulkanHeightmap.hpp(129,22): warning C4305: 'initializing':
 truncation from 'double' to 'float' [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\src\VulkanIn
finiteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\TerrainChunk.cpp(87,16): warning C4244: 'initializing': conv
ersion from 'float' to 'int', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windo
ws\src\VulkanInfiniteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\TerrainChunk.cpp(88,16): warning C4244: 'initializing': conv
ersion from 'float' to 'int', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windo
ws\src\VulkanInfiniteProceduralTerrain.vcxproj]
  VulkanContext.cpp
  main.cpp
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\base\VulkanTexture.hpp(830,57): warning C4267: '=': conversion f
rom 'size_t' to 'uint32_t', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows
\src\VulkanInfiniteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\base\VulkanHeightmap.hpp(129,22): warning C4305: 'initializing':
 truncation from 'double' to 'float' [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\src\VulkanIn
finiteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\main.cpp(433,21): error C2039: 'getHeightAndRandomValue': is
 not a member of 'InfiniteTerrain' [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\src\VulkanInfi
niteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\main.cpp(474,39): warning C4018: '>': signed/unsigned mismat
ch [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\src\VulkanInfiniteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\main.cpp(497,47): warning C4018: '>': signed/unsigned mismat
ch [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\src\VulkanInfiniteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\main.cpp(619,24): warning C4267: '=': conversion from 'size_
t' to 'int', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\src\VulkanInfi
niteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\main.cpp(625,25): warning C4267: '=': conversion from 'size_
t' to 'int', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\src\VulkanInfi
niteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\main.cpp(688,4): error C2659: '=': function as left operand
[C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\src\VulkanInfiniteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\main.cpp(1194,19): warning C4553: '==': result of expression
 not used; did you intend '='? [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\src\VulkanInfinite
ProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\main.cpp(2219,26): warning C4244: 'initializing': conversion
 from 'float' to 'int', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\src
\VulkanInfiniteProceduralTerrain.vcxproj]
C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\src\main.cpp(2220,26): warning C4244: 'initializing': conversion
 from 'float' to 'int', possible loss of data [C:\Users\<user>\Desktop\VulkanInfiniteProceduralTerrain\build\windows\src
\VulkanInfiniteProceduralTerrain.vcxproj]
  Generating Code...

**Comment by SaschaWillems** [comment]
Please read [this](https://github.com/SaschaWillems/VulkanInfiniteProceduralTerrain?tab=readme-ov-file#about)

tl;dr: I don't guarantee that this is working. It's a personal playground.
